### ISSUE TITLE
Option to specify the remote file name

### ISSUE THREAD

**Issue by nthiery** [tavinus/cloudsend.sh#2]
Use case: I have a file `foo.txt` and would like to drop it under the name `bar,txt`

Potential syntax:

    cloudsend foo.txt --remote-file-name bar.txt <folderlink>

**Comment by tavinus** [comment]
Hi.  
  
~I have [pushed a new release (v0.1.7)](https://github.com/tavinus/cloudsend.sh/releases) with the `-r` / `--rename` option to do that.~

**EDIT:**  Ok, I have changed a lot of things that I wanted and also made the option/parameters parsing more robust. [I have released it as v2.0.0](https://github.com/tavinus/cloudsend.sh/releases).  
  
---

**Cloudsend v2 changed the way password parsing works.**  
Cloudsend 0.x.x used the `-p` parameter for the Environment password (changed to `-e` in v2+).  
Please use EITHER `-e` OR `-p`, but not both. The last to be called will be used.  
  
 - **Env Pass** *>* Set the variable `CLOUDSEND_PASSWORD='MySecretPass'` and use the option `-e`
 - **Param Pass** *>* Send the password as a parameter with `-p <password>`

---

**I still need testers, so...
Can you please test it and report back?**

**Comment by nthiery** [comment]
> I have [pushed a new release (v0.1.7)](https://github.com/tavinus/cloudsend.sh/releases) with the `-r` / `--rename` option to do that.  
>   
> Can you please test it and report back?

Cool, it worked flawlessly! Thanks a lot!

**Comment by nthiery** [comment]
Note: I tested with an anonymous file drop; so that did not exercise the password passing options.

**Comment by nthiery** [comment]
Now that the -r option is there, this made me wonder whether one could pipe in the input file?
Something like:

    echo coucou | cloudsend -r <filename> <url>

Typical use case: creating a remote archive on the fly:

    tar -c foo/ | cloudsend.sh -r foo.tar <url>

I can create a separate issue if you prefer.

**Comment by tavinus** [comment]
I don't think you can pipe it like that.   
  
The way this script is implemented, we need a file to send to curl as the origin and then we can use the same name or another one at the destination.  
  
For now, you will need to save the file, send it, and then delete it (in your script).  
  
So we would need to find a way to pipe the content of the file as the origin to curl. I think this is possible, but would change a lot the curl call. 

So, we would probably need a different call for this use case, which may be ok. The first step is to have an example curl call with the content being piped like that.    
  
There is usually a flag to trigger pipe input. Most times a single dash at the end is used ( `-` ).  
  
I am not sure I will have the time to try this soon. Would be easier if you can come up with a curl example.  

---

EDIT: Curl is used with `-T <file>` and the manual says
![image](https://user-images.githubusercontent.com/8039413/81991015-b185bc00-9616-11ea-8e82-4054ea83b0ad.png)

I will try to see if this is already working when using `-` and/or `.` as input file. I will need to add a check for the destination name as well though, and maybe even rename the option.

**Comment by tavinus** [comment]
Hi. I have [released a new version](https://github.com/tavinus/cloudsend.sh/releases) (2.1.0) which can both send multiple files (using globs) and also send piped content (using `-` OR `.` as input file name).  

---

**Here are the release notes:**
 - [x] Adds stdin as input option (piped content)
      - uses either `-` or `.` as input file name
      - requires `-r <filename>`
 - [x] Allows to use globbing at the input
      - can send multiple files by using globs
      - requires `-g` option
      - not compatible with `-r <name>` option (rename)
 - [x] Adds new checks and parsing
 - [x] Cleaned up unused code
 - [x] Updated readme and `--help`

[There is more info on the main Readme file](https://github.com/tavinus/cloudsend.sh).

**Comment by nthiery** [comment]
Cool! I just tried:

    echo coucou | /opt/cloudsend.sh/cloudsend.sh . -r truc.md <url>

The file truc.md is indeed created, but empty. Am I doing something wrong?

**Comment by tavinus** [comment]
That is weird, I just tried echoing without problems.  
Both `-` and `.` worked fined.  
Both quoted and unquoted strings, need quotes if sending multiple lines/spaces/etc.  
But `echo Test` also worked just fine.   
  
With `.`
```
echo $'testing\nSuper Test\nTestAgain' | ./cloudsend.sh . https://cloud.mydomain.net/s/TxWnFJ7Ln3X5yHs -r testPipe4.txt -p MyPass123
```
With `-`
```
echo $'testing\nSuper Test\nTestAgain' | ./cloudsend.sh - https://cloud.mydomain.net/s/TxWnFJ7Ln3X5yHs -r testPipe5.txt -p MyPass123
```
And the files
![image](https://user-images.githubusercontent.com/8039413/82096149-ccb80080-96d6-11ea-816a-db90fc1b503a.png)
![image](https://user-images.githubusercontent.com/8039413/82096168-d6416880-96d6-11ea-96c4-6f0da20ab53f.png)  
  
**EDIT** I was wondering if having the `-r <name>`  between `.` AND `URL` was a problem, but I just tested it and it also went fine.
